### ISSUE TITLE
Frequent crash in etcd due to null pointer starting in version 5.17

### ISSUE THREAD

**Issue by Deltik** [sensu/sensu-go#3550]
## Expected Behavior
Retain behavior from version 5.16.1: No crash every 5 to 6 minutes.

## Current Behavior
Sensu Go Backend version 5.17.0 to 5.17.1 crashes roughly every 6 minutes with a segmentation fault.

## Steps to Reproduce (for bugs)
I wrote a SaltStack state file, which automatically deploys `sensu-go-backend`.  Here are all the relevant files, which you can use to deploy the crashing `sensu-go-backend` yourself:

`./monitoring/base.sls`:
```yaml
{% from "monitoring/map.jinja" import sensu_repo with context %}

{% if 'Debian' in grains.os_family %}
Install GnuPG for apt-key:
  pkg.installed:
    - name: gpg
{% endif %}

Sensu repository:
  pkgrepo.managed:
    - name: {{ sensu_repo.name }}
    - enabled: True
    - humanname: sensu_stable
{% if 'Debian' in grains.os_family %}
    - file: {{ sensu_repo.file }}
    - key_url: https://packagecloud.io/sensu/stable/gpgkey
{% endif %}
    - baseurl: {{ sensu_repo.baseurl }}
    - gpgkey: https://packagecloud.io/sensu/stable/gpgkey
    - gpgcheck: 0
    - repo_gpgcheck: 1
```

`./monitoring/server.sls` (Docker):
```yaml
include:
  - monitoring.base

{% from "monitoring/map.jinja" import docker with context %}

{% if 'CentOS' in grains.os %}
Docker repository:
  pkgrepo.managed:
    - name: docker-ce
    - enabled: True
    - humanname: docker-ce-stable
    - baseurl: https://download.docker.com/linux/centos/$releasever/$basearch/stable
    - gpgkey: https://download.docker.com/linux/centos/gpg
    - gpgcheck: 1
{% endif %}

Install Docker:
  pkg.installed:
    - pkgs:
      - {{ docker.package_name }}
      - {{ docker.python_bindings_package_name }}

Start and enable Docker:
  service.running:
    - name: docker
    - enable: True

Pull Sensu Docker image:
  docker_image.present:
    - name: sensu/sensu
    - tag: latest

Start Sensu server:
  docker_container.running:
    - name: sensu-backend
    - image: sensu/sensu
    - command: sensu-backend start --state-dir /var/lib/sensu/sensu-backend --log-level warn
    - binds:
      -  /etc/sensu:/etc/sensu
      -  /var/lib/sensu:/var/lib/sensu
    - port_bindings:
      - 3000:3000
      - 8080:8080
      - 8081:8081
    - restart_policy: unless-stopped
```

`./monitoring/server.sls` (apt):
```yaml
include:
  - monitoring.base

/etc/sensu:
  file.directory:
    - name: /etc/sensu
    - user: sensu
    - group: sensu
    - mode: 0755
    - watch_in:
      - service: sensu-backend

/etc/sensu/backend.yml:
  file.managed:
    - name: /etc/sensu/backend.yml
    - user: sensu
    - group: sensu
    - mode: 0600
    - source:
      - salt://monitoring/files/etc/sensu/backend.yml
    - template: jinja
    - watch_in:
      - service: sensu-backend

/var/lib/sensu:
  file.directory:
    - name: /var/lib/sensu
    - user: sensu
    - group: sensu
    - recurse:
      - user
      - group
      - mode
    - watch_in:
      - service: sensu-backend

Install sensu-backend:
  pkg.installed:
    - pkgs:
      - sensu-go-backend

Start Sensu server:
  service.running:
    - name: sensu-backend
    - enable: True

Install sensuctl:
  pkg.installed:
    - pkgs:
      - sensu-go-cli
```

`./monitoring/map.jinja`:
```jinja
{% set docker = salt['grains.filter_by']({
  'Debian': {
    'package_name': 'docker.io',
    'python_bindings_package_name': 'python3-docker',
  },
  'RedHat': {
    'package_name': 'docker-ce',
    'python_bindings_package_name': 'python36-docker',
  }
}, grain='os_family') %}

{% set sensu_repo = salt['grains.filter_by']({
  'Debian': {
    'name': 'deb https://packagecloud.io/sensu/stable/ubuntu/ ' + salt['grains.get']('oscodename') + ' main',
    'baseurl': None,
    'file': '/etc/apt/sources.list.d/sensu.list',
  },
  'RedHat': {
    'name': 'sensu',
    'baseurl': 'https://packagecloud.io/sensu/stable/el/$releasever/$basearch',
    'file': None,
  }
}, grain='os_family') %}
```

`./monitoring/files/etc/sensu/backend.yml`:
```yaml
---
state-dir: "{{ salt['pillar.get']('sensu:backend:state_dir', '/var/lib/sensu/sensu-backend') }}"
log-level: "{{ salt['pillar.get']('sensu:backend:log_level', 'warn') }}"
```

## Context
Here is the stack trace of one of the many crashes:

```
Feb 07 09:22:20 monitoring sensu-backend[32594]: panic: runtime error: invalid memory address or nil pointer dereference
Feb 07 09:22:20 monitoring sensu-backend[32594]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12e508f]
Feb 07 09:22:20 monitoring sensu-backend[32594]: goroutine 172 [running]:
Feb 07 09:22:20 monitoring sensu-backend[32594]: github.com/coreos/bbolt.(*Tx).Bucket(...)
Feb 07 09:22:20 monitoring sensu-backend[32594]:         /go/pkg/mod/github.com/coreos/bbolt@v1.3.3/tx.go:101
Feb 07 09:22:20 monitoring sensu-backend[32594]: github.com/coreos/etcd/mvcc/backend.(*readTx).UnsafeRange(0xc0002e19a0, 0x28cc090, 0x3, 0x3, 0xc00148dde0, 0x11, 0x12, 0x0, 0x0, 0x0, ...)
Feb 07 09:22:20 monitoring sensu-backend[32594]:         /go/pkg/mod/github.com/coreos/etcd@v3.3.17+incompatible/mvcc/backend/read_tx.go:75 +0x66f
Feb 07 09:22:20 monitoring sensu-backend[32594]: github.com/coreos/etcd/mvcc.(*storeTxnRead).rangeKeys(0xc000c7aba0, 0xc00148dd00, 0x13, 0x20, 0xc00148dd40, 0x12, 0x20, 0xa3, 0x0, 0x0, ...)
Feb 07 09:22:20 monitoring sensu-backend[32594]:         /go/pkg/mod/github.com/coreos/etcd@v3.3.17+incompatible/mvcc/kvstore_txn.go:140 +0x287
Feb 07 09:22:20 monitoring sensu-backend[32594]: github.com/coreos/etcd/mvcc.(*storeTxnRead).Range(0xc000c7aba0, 0xc00148dd00, 0x13, 0x20, 0xc00148dd40, 0x12, 0x20, 0x0, 0x0, 0x0, ...)
Feb 07 09:22:20 monitoring sensu-backend[32594]:         /go/pkg/mod/github.com/coreos/etcd@v3.3.17+incompatible/mvcc/kvstore_txn.go:45 +0xaf
Feb 07 09:22:20 monitoring sensu-backend[32594]: github.com/coreos/etcd/mvcc.(*metricsTxnWrite).Range(0xc000c7abd0, 0xc00148dd00, 0x13, 0x20, 0xc00148dd40, 0x12, 0x20, 0x0, 0x0, 0x0, ...)
Feb 07 09:22:20 monitoring sensu-backend[32594]:         /go/pkg/mod/github.com/coreos/etcd@v3.3.17+incompatible/mvcc/metrics_txn.go:38 +0xaf
Feb 07 09:22:20 monitoring sensu-backend[32594]: github.com/coreos/etcd/etcdserver.(*applierV3backend).Range(0xc0002f89a0, 0x1d37f80, 0xc000c7abd0, 0xc0002773b0, 0x0, 0x0, 0x0)
Feb 07 09:22:20 monitoring sensu-backend[32594]:         /go/pkg/mod/github.com/coreos/etcd@v3.3.17+incompatible/etcdserver/apply.go:269 +0x13f
Feb 07 09:22:20 monitoring sensu-backend[32594]: github.com/coreos/etcd/etcdserver.(*EtcdServer).Range.func3()
Feb 07 09:22:20 monitoring sensu-backend[32594]:         /go/pkg/mod/github.com/coreos/etcd@v3.3.17+incompatible/etcdserver/v3_server.go:103 +0x60
Feb 07 09:22:20 monitoring sensu-backend[32594]: github.com/coreos/etcd/etcdserver.(*EtcdServer).doSerialize(0xc000230f00, 0x1d37a80, 0xc0002ba940, 0xc001163330, 0xc001163348, 0x7ff3eec28fff, 0x3)
Feb 07 09:22:20 monitoring sensu-backend[32594]:         /go/pkg/mod/github.com/coreos/etcd@v3.3.17+incompatible/etcdserver/v3_server.go:542 +0xec
Feb 07 09:22:20 monitoring sensu-backend[32594]: github.com/coreos/etcd/etcdserver.(*EtcdServer).Range(0xc000230f00, 0x1d37a80, 0xc0002ba940, 0xc0002773b0, 0x0, 0x0, 0x0)
Feb 07 09:22:20 monitoring sensu-backend[32594]:         /go/pkg/mod/github.com/coreos/etcd@v3.3.17+incompatible/etcdserver/v3_server.go:104 +0x1be
Feb 07 09:22:20 monitoring sensu-backend[32594]: github.com/coreos/etcd/etcdserver/api/v3rpc.(*kvServer).Range(0xc0002baa40, 0x1d37a80, 0xc0002ba940, 0xc0002773b0, 0x7ff3f0c3cf80, 0x0, 0x414587)
Feb 07 09:22:20 monitoring sensu-backend[32594]:         /go/pkg/mod/github.com/coreos/etcd@v3.3.17+incompatible/etcdserver/api/v3rpc/key.go:52 +0x78
Feb 07 09:22:20 monitoring sensu-backend[32594]: github.com/coreos/etcd/proxy/grpcproxy/adapter.(*kvs2kvc).Range(0xc000a680e0, 0x1d37a80, 0xc0002ba940, 0xc0002773b0, 0x0, 0x0, 0x0, 0x7ff3f0c3cf80, 0x413ed3, 0xc000b2db70)
Feb 07 09:22:20 monitoring sensu-backend[32594]:         /go/pkg/mod/github.com/coreos/etcd@v3.3.17+incompatible/proxy/grpcproxy/adapter/kv_client_adapter.go:32 +0x51
Feb 07 09:22:20 monitoring sensu-backend[32594]: github.com/coreos/etcd/clientv3.(*kv).Do(0xc000a5c780, 0x1d37a80, 0xc0002ba940, 0x1, 0xc00148dd00, 0x13, 0x20, 0xc00148dd40, 0x12, 0x20, ...)
Feb 07 09:22:20 monitoring sensu-backend[32594]:         /go/pkg/mod/github.com/coreos/etcd@v3.3.17+incompatible/clientv3/kv.go:149 +0x1e4
Feb 07 09:22:20 monitoring sensu-backend[32594]: github.com/coreos/etcd/clientv3.(*kv).Get(0xc000a5c780, 0x1d37a80, 0xc0002ba940, 0xc00148dcc0, 0x13, 0xc000635fa0, 0x3, 0x4, 0x1786ee0, 0xc001163a88, ...)
Feb 07 09:22:20 monitoring sensu-backend[32594]:         /go/pkg/mod/github.com/coreos/etcd@v3.3.17+incompatible/clientv3/kv.go:119 +0x1c7
Feb 07 09:22:20 monitoring sensu-backend[32594]: github.com/sensu/sensu-go/backend/store/etcd.List(0x1d37a80, 0xc0002ba940, 0xc0002f0000, 0xc001163ba8, 0x1559880, 0xc000635f40, 0xc001163bc0, 0x0, 0x0)
Feb 07 09:22:20 monitoring sensu-backend[32594]:         /go/pkg/mod/github.com/sensu/sensu-go@v0.0.0-20200131164840-40b1d5938251/backend/store/etcd/store.go:204 +0x5f9
Feb 07 09:22:20 monitoring sensu-backend[32594]: github.com/sensu/sensu-go/backend/store/cache.getResources(0x1d37a80, 0xc0002ba940, 0xc0002f0000, 0x1d4b180, 0xc000bc4000, 0x1, 0x1, 0x0, 0xc001163d60, 0x4414b4)
Feb 07 09:22:20 monitoring sensu-backend[32594]:         /go/pkg/mod/github.com/sensu/sensu-go@v0.0.0-20200131164840-40b1d5938251/backend/store/cache/cache.go:117 +0x26f
Feb 07 09:22:20 monitoring sensu-backend[32594]: github.com/sensu/sensu-go/backend/store/cache.(*Resource).rebuild(0xc000c98300, 0x1d37a80, 0xc0002ba940, 0x0, 0x0, 0x0)
Feb 07 09:22:20 monitoring sensu-backend[32594]:         /go/pkg/mod/github.com/sensu/sensu-go@v0.0.0-20200131164840-40b1d5938251/backend/store/cache/cache.go:253 +0x103
Feb 07 09:22:20 monitoring sensu-backend[32594]: github.com/sensu/sensu-go/backend/store/cache.(*Resource).start(0xc000c98300, 0x1d37a80, 0xc0002ba940)
Feb 07 09:22:20 monitoring sensu-backend[32594]:         /go/pkg/mod/github.com/sensu/sensu-go@v0.0.0-20200131164840-40b1d5938251/backend/store/cache/cache.go:239 +0x161
Feb 07 09:22:20 monitoring sensu-backend[32594]: created by github.com/sensu/sensu-go/backend/store/cache.New
Feb 07 09:22:20 monitoring sensu-backend[32594]:         /go/pkg/mod/github.com/sensu/sensu-go@v0.0.0-20200131164840-40b1d5938251/backend/store/cache/cache.go:153 +0x158
```

I attempted to work around this by having systemd restart `sensu-backend.service` every time it crashed, but sometimes the backend just doesn't respond even though the service is running.  As the workaround wasn't acceptable, I've downgraded to version 5.16.1 for the time being.

Apparently not related?: #3546

## Your Environment
* **sensu-backend version:** ~> 5.17.0
* **Installation method:** affects both the official Docker container and the official Ubuntu 18.04 repository (`deb https://packagecloud.io/sensu/stable/ubuntu bionic main`)
* **Operating system and version:** Ubuntu 18.04 on x86_64

**Comment by echlebek** [comment]
@Deltik I believe this should be fixed in 5.18.1. Would you mind giving that a try and reporting back success or failure? Thanks!

**Comment by Deltik** [comment]
There hasn't been a crash since the upgrade.  I think version 5.18.1 solves this issue!
